src/panels/logbook/ha-logbook.ts

```diff
@@ -254,6 +276,97 @@ class HaLogbook extends LitElement {
     this._savedScrollPos = (e.target as HTMLDivElement).scrollTop;
   }
 
+  private _formatEventBy = (item: LogbookEntry, seenEntities: string[]) => {
+    if (item.context_event_type === "call_service") {
+      return `${this.hass.localize("ui.components.logbook.from_service")} ${
+        item.context_domain
+      }.${item.context_service}`;
+    }
+    if (
+      item.context_event_type === "automation_triggered" &&
+      item.context_entity_id
+    ) {
+      if (seenEntities.includes(item.context_entity_id)) {
+        return ``;
+      }
+      seenEntities.push(item.context_entity_id);
+      return html`${this.hass.localize("ui.components.logbook.from_automation")}
+        <a
+          href="#"
+          @click=${this._entityClicked}
+          .entityId=${item.context_entity_id}
+          class="name"
+          >${item.context_name}</a
+        >`;
+    }
+    if (item.context_name) {
+      return `${this.hass.localize("ui.components.logbook.from")} ${
+        item.context_name
+      }`;
+    }
+    if (item.context_event_type === "state_changed") {
+      return ``;
+    }
+    if (item.context_event_type! in EVENT_LOCALIZE_MAP) {
+      return `${this.hass.localize(
+        `ui.components.logbook.${EVENT_LOCALIZE_MAP[item.context_event_type!]}`
+      )}`;
+    }
+    return `${this.hass.localize(
+      "ui.components.logbook.from"
+    )} ${this.hass.localize("ui.components.logbook.event")} ${
+      item.context_event_type
+    }`;
+  };
+
+  private _formatMessageWithPossibleEntity(
+    message: string,
+    seenEntities: string[],
+    forEntityId?: string,
+    localizePrefix?: string
+  ) {
+    const matches = message.match(/state of ([^ ]+)/);
```

Comment:
I don't like the specific test matching.

As we have more integrations opt into logbook events, this is going to get even more arbitrary.  It might be better to build something that scans the string for entity ids that exist in the state machine and replace them.  

Also it could be caused by a trigger as well.

Reply:
I hand rolled this since I didn't find anything already that did what I wanted with an '`html`'